src/balancer-strategy-manager/BalancerV2WeightedPoolStrategyManager.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {OwnableWithGuardian} from 'solidity-utils/contracts/access-control/OwnableWithGuardian.sol';
+import {Rescuable} from 'solidity-utils/contracts/utils/Rescuable.sol';
+import {RescuableBase, IRescuableBase} from 'solidity-utils/contracts/utils/RescuableBase.sol';
+
+import {IBalancerPool} from './balancer-v2/IBalancerPool.sol';
+import {IBalancerVault, IAsset} from './balancer-v2/IBalancerVault.sol';
+import {WeightedPoolUserData} from './balancer-v2/WeightedPoolUserData.sol';
+import {IBalancerStrategyManager} from './IBalancerStrategyManager.sol';
+
+contract BalancerV2WeightedPoolStrategyManager is
+  IBalancerStrategyManager,
+  OwnableWithGuardian,
+  Rescuable
+{
+  using SafeERC20 for IERC20;
+
+  bytes32 public immutable POOL_ID;
```

Comment:
Please add natspec on Interace and @inheritdoc here for all of these, also please rearrange.